new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Voyager.java

```diff
@@ -0,0 +1,155 @@
+                                           String domainNamespace,
+                                           String domainUid,
+                                           Map<String, Integer> clusterNameMsPortMap) {
+
+    final String ingressApiVersion = "extensions/v1beta1";
+    final String ingressKind = "Ingress";
+    final String ingressType = "NodePort";
+    final String ingressAffinity = "cookie";
+    final String ingressClass = "voyager";
+
+    // set the annotations for Voyager
+    HashMap<String, String> annotation = new HashMap<>();
+    annotation.put("ingress.appscode.com/type", ingressType);
+    annotation.put("ingress.appscode.com/affinity", ingressAffinity);
+    annotation.put("kubernetes.io/ingress.class", ingressClass);
+
+    List<String> ingressHostList = new ArrayList<>();
+    ArrayList<ExtensionsV1beta1IngressRule> ingressRules = new ArrayList<>();
+    clusterNameMsPortMap.forEach((clusterName, managedServerPort) -> {
+      // set the http ingress paths
+      ExtensionsV1beta1HTTPIngressPath httpIngressPath = new ExtensionsV1beta1HTTPIngressPath()
+          .path(null)
+          .backend(new ExtensionsV1beta1IngressBackend()
+              .serviceName(domainUid + "-cluster-" + clusterName.toLowerCase().replace("_", "-"))
+              .servicePort(new IntOrString(managedServerPort))
+          );
+      ArrayList<ExtensionsV1beta1HTTPIngressPath> httpIngressPaths = new ArrayList<>();
+      httpIngressPaths.add(httpIngressPath);
+
+      // set the ingress rule
+      String ingressHost = domainUid + "." + clusterName + ".org";
+      ExtensionsV1beta1IngressRule ingressRule = new ExtensionsV1beta1IngressRule()
+          .host(ingressHost)
+          .http(new ExtensionsV1beta1HTTPIngressRuleValue()
+              .paths(httpIngressPaths));
+
+      ingressRules.add(ingressRule);
+      ingressHostList.add(ingressHost);
+    });
+
+    // set the ingress
+    ExtensionsV1beta1Ingress ingress = new ExtensionsV1beta1Ingress()
+        .apiVersion(ingressApiVersion)
+        .kind(ingressKind)
+        .metadata(new V1ObjectMeta()
+            .name(ingressName)
+            .namespace(domainNamespace)
+            .annotations(annotation))
+        .spec(new ExtensionsV1beta1IngressSpec()
+            .rules(ingressRules));
+
+    // create the ingress
+    try {
+      Kubernetes.createIngress(domainNamespace, ingress);
+    } catch (ApiException apex) {
+      logger.severe("got ApiException while calling createIngress: {0}", apex.getResponseBody());
+      return null;
+    }
+    return ingressHostList;
+  }
+
+  /**
+   * List all of the ingresses in the specified namespace.
+   *
+   * @param namespace the namespace to which the ingresses belong
+   * @return a list of ingress names in the namespace
+   * @throws ApiException if Kubernetes client API call fails
+   */
+  public static List<String> listIngresses(String namespace) throws ApiException {
```

Comment:
I see the same code in Nginx as well for listIngresses() and slightly different for createIngress() methods. Can Ingress be pulled out as another impl class instead of duplicate code?

Reply:
agreed. but we need to discuss how to

Reply:
because we currently have nginx or voyager hard-coded. We need a name convention in order to share same method, IMO

Reply:
listIngress should be the same no matter which ingress controller is used.

Reply:
Only annotations are different for each ingress controller. You can create a class Ingress which will have createIngress and listIngresses methods. createIngress takes annotations as one of the parameter. Nginx and Voyager can call Ingress class for creareIngress.

Reply:
That's Vanaja's point - we may need to consolidate it to a class that is not named Nginx or Voyager. It may cause confusion if Voyager calls Niginx.listIngress, right?

Reply:
That's what I am thinking @vanajamukkara

Reply:
fixed